### PR TITLE
ListBooks fails with listSize > 10

### PR DESCRIPTION
**Pull request by ramon-reichert**
The 'test results' do not show exactly what is the error, but a listSize from 11 causes a fail. I'm still searching it.

### CODE REVIEW

File: cmd/api/database_test.go [hunk 1]
```diff
@@ -97,18 +101,36 @@ func TestListBooks(t *testing.T) {
 		is := is.New(t)
 
 		// Write the List Books test here.
-		// ...
-		is.Fail()
+		returnedBooks, err := listBooks()
```
Comment (cesarFuhr) [inline cmd/api/database_test.go:104]: Lets make `listBooks` return an empty slice (`[]Book{}`) instead of `nil` when there is no error but no book either.

**Comment by ramon-reichert** [comment]
Is there something wrong with the user of the test database? Look at  the logs of the container (the end, especially):

2023-08-07 14:31:28 books-service-test-db-1  | The files belonging to this database system will be owned by user "postgres".
2023-08-07 14:31:28 books-service-test-db-1  | This user must also own the server process.
2023-08-07 14:31:28 books-service-test-db-1  | 
2023-08-07 14:31:28 books-service-test-db-1  | The database cluster will be initialized with locale "en_US.utf8".
2023-08-07 14:31:28 books-service-test-db-1  | The default database encoding has accordingly been set to "UTF8".
2023-08-07 14:31:28 books-service-test-db-1  | The default text search configuration will be set to "english".
2023-08-07 14:31:28 books-service-test-db-1  | 
2023-08-07 14:31:28 books-service-test-db-1  | Data page checksums are disabled.
2023-08-07 14:31:28 books-service-test-db-1  | 
2023-08-07 14:31:28 books-service-test-db-1  | fixing permissions on existing directory /var/lib/postgresql/data ... ok
2023-08-07 14:31:28 books-service-test-db-1  | creating subdirectories ... ok
2023-08-07 14:31:28 books-service-test-db-1  | selecting dynamic shared memory implementation ... posix
2023-08-07 14:31:28 books-service-test-db-1  | selecting default max_connections ... 100
2023-08-07 14:31:28 books-service-test-db-1  | selecting default shared_buffers ... 128MB
2023-08-07 14:31:28 books-service-test-db-1  | selecting default time zone ... Etc/UTC
2023-08-07 14:31:28 books-service-test-db-1  | creating configuration files ... ok
2023-08-07 14:31:28 books-service-test-db-1  | running bootstrap script ... ok
2023-08-07 14:31:29 books-service-test-db-1  | performing post-bootstrap initialization ... ok
2023-08-07 14:31:31 books-service-test-db-1  | initdb: warning: enabling "trust" authentication for local connections
2023-08-07 14:31:31 books-service-test-db-1  | You can change this by editing pg_hba.conf or using the option -A, or
2023-08-07 14:31:31 books-service-test-db-1  | --auth-local and --auth-host, the next time you run initdb.
2023-08-07 14:31:31 books-service-test-db-1  | syncing data to disk ... ok
2023-08-07 14:31:31 books-service-test-db-1  | 
2023-08-07 14:31:31 books-service-test-db-1  | 
2023-08-07 14:31:31 books-service-test-db-1  | Success. You can now start the database server using:
2023-08-07 14:31:31 books-service-test-db-1  | 
2023-08-07 14:31:31 books-service-test-db-1  |     pg_ctl -D /var/lib/postgresql/data -l logfile start
2023-08-07 14:31:31 books-service-test-db-1  | 
2023-08-07 14:31:31 books-service-test-db-1  | waiting for server to start....2023-08-07 17:31:31.308 UTC [48] LOG:  starting PostgreSQL 14.6 (Debian 14.6-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
2023-08-07 14:31:31 books-service-test-db-1  | 2023-08-07 17:31:31.376 UTC [48] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-08-07 14:31:31 books-service-test-db-1  | 2023-08-07 17:31:31.578 UTC [49] LOG:  database system was shut down at 2023-08-07 17:31:29 UTC
2023-08-07 14:31:31 books-service-test-db-1  | 2023-08-07 17:31:31.659 UTC [48] LOG:  database system is ready to accept connections
2023-08-07 14:31:31 books-service-test-db-1  |  done
2023-08-07 14:31:31 books-service-test-db-1  | server started
2023-08-07 14:31:33 books-service-test-db-1  | CREATE DATABASE
2023-08-07 14:31:33 books-service-test-db-1  | 
2023-08-07 14:31:33 books-service-test-db-1  | 
2023-08-07 14:31:33 books-service-test-db-1  | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
2023-08-07 14:31:33 books-service-test-db-1  | 
2023-08-07 14:31:33 books-service-test-db-1  | waiting for server to shut down...2023-08-07 17:31:33.218 UTC [48] LOG:  received fast shutdown request
2023-08-07 14:31:33 books-service-test-db-1  | .2023-08-07 17:31:33.282 UTC [48] LOG:  aborting any active transactions
2023-08-07 14:31:33 books-service-test-db-1  | 2023-08-07 17:31:33.287 UTC [48] LOG:  background worker "logical replication launcher" (PID 55) exited with exit code 1
2023-08-07 14:31:33 books-service-test-db-1  | 2023-08-07 17:31:33.287 UTC [50] LOG:  shutting down
2023-08-07 14:31:33 books-service-test-db-1  | 2023-08-07 17:31:33.708 UTC [48] LOG:  database system is shut down
2023-08-07 14:31:33 books-service-test-db-1  |  done
2023-08-07 14:31:33 books-service-test-db-1  | server stopped
2023-08-07 14:31:33 books-service-test-db-1  | 
2023-08-07 14:31:33 books-service-test-db-1  | PostgreSQL init process complete; ready for start up.
2023-08-07 14:31:33 books-service-test-db-1  | 
2023-08-07 14:31:33 books-service-test-db-1  | 2023-08-07 17:31:33.876 UTC [1] LOG:  starting PostgreSQL 14.6 (Debian 14.6-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
2023-08-07 14:31:33 books-service-test-db-1  | 2023-08-07 17:31:33.876 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2023-08-07 14:31:33 books-service-test-db-1  | 2023-08-07 17:31:33.877 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2023-08-07 14:31:33 books-service-test-db-1  | 2023-08-07 17:31:33.993 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2023-08-07 14:31:34 books-service-test-db-1  | 2023-08-07 17:31:34.137 UTC [63] LOG:  database system was shut down at 2023-08-07 17:31:33 UTC
2023-08-07 14:31:34 books-service-test-db-1  | 2023-08-07 17:31:34.213 UTC [1] LOG:  database system is ready to accept connections
2023-08-07 14:31:48 books-service-test-db-1  | 2023-08-07 17:31:48.948 UTC [79] FATAL:  role "postgres" does not exist
2023-08-07 14:32:09 books-service-test-db-1  | 2023-08-07 17:32:09.298 UTC [89] FATAL:  role "postgres" does not exist
2023-08-07 14:32:29 books-service-test-db-1  | 2023-08-07 17:32:29.658 UTC [98] FATAL:  role "postgres" does not exist
2023-08-07 14:32:49 books-service-test-db-1  | 2023-08-07 17:32:49.923 UTC [107] FATAL:  role "postgres" does not exist
2023-08-07 14:33:10 books-service-test-db-1  | 2023-08-07 17:33:10.269 UTC [117] FATAL:  role "postgres" does not exist
2023-08-07 14:33:30 books-service-test-db-1  | 2023-08-07 17:33:30.620 UTC [125] FATAL:  role "postgres" does not exist
2023-08-07 14:33:50 books-service-test-db-1  | 2023-08-07 17:33:50.953 UTC [135] FATAL:  role "postgres" does not exist

**Comment by cesarFuhr** [comment]
No, its the slice ordering. 
The database is ordering them based on their name, but it compares each character individually not the actual numbers. So `Book 9` ends up being after `Books 10`... but the created books slice stores them in the created order.
You have a few options:
- You could left pad the book numbers like this `Book 01` ... `Book 10`.
- You could order your created books slice after storing them with `sort.Slice()`.

**Review by cesarFuhr — CHANGES_REQUESTED** [review]

File: docker-compose.test.yml
Comment (ramon-reichert) [inline docker-compose.test.yml:1]: This change stopped the healthcheck error that was logging "FATAL: role "postgres" does not exist", but I've notice this:
"2023-08-07 14:31:28 books-service-test-db-1 | The files belonging to this database system will be owned by user "postgres". 
Shouldn't the user be "root"?

File: cmd/api/database.go [hunk 2]
```diff
@@ -95,7 +95,7 @@ func listBooks() ([]Book, error) {
 		return nil, fmt.Errorf("listing all books on db: %w", err)
 	}
 	defer rows.Close()
-	var bookslist []Book
+	var bookslist []Book = []Book{}
```
Comment (cesarFuhr) [inline cmd/api/database.go]: Since you are setting a value, do it this way:
```go
bookslist := []Book{}

File: cmd/api/database_test.go [hunk 3]
```diff
@@ -97,18 +101,36 @@ func TestListBooks(t *testing.T) {
...
+		is.Equal(returnedBooks, []Book{})
 	})
 
 	t.Run("List books without errors", func(t *testing.T) {
 		is := is.New(t)
 
 		// Setting up, creating books to be listed.
-		// ...
+		var bookslist []Book
+		listSize := 100
```
Comment (cesarFuhr) [inline cmd/api/database_test.go]: I like the spirit of this, but there is no need to create 100 books to do the test. It's essentially the same as testing with 11.

Let's keep it around 11 or 20.

File: docker-compose.test.yml
Comment (cesarFuhr) [reply docker-compose.test.yml:1]: It should, nice catch! 🚀

**Review by cesarFuhr — CHANGES_REQUESTED** [review]
Fix those little things, other than that LGTM!

File: cmd/api/database_test.go (hunk 3 above)
Comment (cesarFuhr) [reply cmd/api/database_test.go]: Tests should run fast.

**Review by cesarFuhr — APPROVED** [review]

**Pull request merged**